### PR TITLE
fix crash while using multi_socket_action

### PR DESCRIPTION
**Pull request by uudiin**

### CODE REVIEW

**Comment by moteus** [comment]
Can you provide test case?

**Comment by moteus** [comment]
I see what your problem. Thanks for bug report.
But I am not sure how to properly fix it. 
I can not find any documentation about when cURL may call callbacks.
I thouth it can call callbacks only during `perform` and `socket_action` calls.
But I found that `timerfunction` can be called during `add_handle` call. 
So now question is can cURL call any easy callbacks in such cases and shuld I set 
Lua state only for multi handle or for all easy also?

```Lua
multi = curl.multi{
  timerfunction = on_curl_timeout;
  socketfunction = on_curl_action;
}

-- here cURL may call e.g. `timerfunction` .
coroutine.wrap(function() multi:add_handle(c) end)()

```

PS. It will be good if you have any simple testcase without any external deps.

**Comment by uudiin** [comment]
I will provide testcase later. it depend on libuv and luv(lua libuv)
it is a example based multi-uv.c in curl source tree (curl/docs/examples/multi-uv.c)

**Comment by moteus** [comment]
I know about this :). And I use this example to test on local machine.
But I think how easily run this on Appveyor.

**Comment by moteus** [comment]
I have similar example [uvwget](/Lua-cURL/Lua-cURLv3/blob/master/examples/cURLv3/uvwget.lua)
I make my PR wich fix this bug. But it also works with coroutines.
Can you test it?

File: test/test_multi_uv.lua [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        if wget.post then
+            if type(wget.post) == "string" then
+                easy:setopt_postfields(wget.post)
+            else
+                easy:setopt_httppost(wget.post)
+            end
+        end
+
+        wget.data = {}
+        easy:setopt_writefunction(function(ud, s)
```
Comment (moteus) [inline test/test_multi_uv.lua]: Just to notice. you can use just `easy:setopt_writefunction(table.insert, wget.data)`

Comment (uudiin) [inline test/test_multi_uv.lua]: Thank you, I got it

**Pull request closed without merging**